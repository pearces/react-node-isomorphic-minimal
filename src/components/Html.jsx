import React from 'react';
import PropTypes from 'prop-types';

const Html = ({
  children,
  title,
  description,
  stylesheets,
  scripts
}) => (
  <html lang="en">
    <head>
      <meta charSet="utf-8" />
      <title>{title}</title>
      <meta name="description" content={description} />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      {stylesheets.map((stylesheet) => <link rel="stylesheet" href={`/static/${stylesheet}`} key={stylesheet} />)}
      <link rel="apple-touch-icon" sizes="180x180" href="/static/apple-touch-icon.png" />
      <link rel="icon" type="image/png" sizes="32x32" href="/static/favicon-32x32.png" />
      <link rel="icon" type="image/png" sizes="16x16" href="/static/favicon-16x16.png" />
    </head>
    <body>
      <div id="app">{children}</div>
      {scripts.map((script) => <script src={`/static/${script}`} key={script} />)}
    </body>
  </html>
);

Html.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  stylesheets: PropTypes.arrayOf(PropTypes.string),
  scripts: PropTypes.arrayOf(PropTypes.string)
};

Html.defaultProps = {
  title: '',
  description: '',
  stylesheets: [],
  scripts: []
};

export default Html;
