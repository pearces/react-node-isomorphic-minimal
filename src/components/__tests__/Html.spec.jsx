import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Html from '../Html';

describe('Html', () => {
  it('matches component snapshot', () => {
    const html = renderer.create(<Html><div /></Html>);
    const tree = html.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with stylesheets', () => {
    const stylesheets = ['foo.css'];
    const html = render(<Html stylesheets={stylesheets}><div /></Html>).container;
    expect(html.querySelectorAll(`link[href*='${stylesheets[0]}']`).length).toEqual(1);
  });

  it('renders with scripts', () => {
    const scripts = ['bar.js'];
    const html = render(<Html scripts={scripts}><div /></Html>).container;
    expect(html.querySelectorAll(`script[src*='${scripts[0]}`).length).toEqual(1);
  });

  it('renders with inline css', () => {
    const styles = 'html {background: "#fff"}';
    render(<Html inlineCss={styles}><div /></Html>);
    expect(document.querySelector('style')).toBeTruthy(); // need to get via document
  });

  it('renders with inline script', () => {
    const scripts = ['console.log("foo")'];
    render(<Html inlineScripts={scripts}><div /></Html>);
    expect(document.querySelector('script')).toBeTruthy(); // need to get via document
  });
});
