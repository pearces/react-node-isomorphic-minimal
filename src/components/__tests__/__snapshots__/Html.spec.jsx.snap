// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html matches component snapshot 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charSet="utf-8"
    />
    <title />
    <meta
      content=""
      name="description"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <link
      href="/static/apple-touch-icon.png"
      rel="apple-touch-icon"
      sizes="180x180"
    />
    <link
      href="/static/favicon-32x32.png"
      rel="icon"
      sizes="32x32"
      type="image/png"
    />
    <link
      href="/static/favicon-16x16.png"
      rel="icon"
      sizes="16x16"
      type="image/png"
    />
  </head>
  <body>
    <div
      id="app"
    >
      <div />
    </div>
  </body>
</html>
`;
